### DISCUSSION TITLE
board = esp32-2432S024C compilation error

### DISCUSSION THREAD

**Discussion by vorosj** [rzeldent/esp32-smartdisplay#110] [Q&A] [answered]
Hi, I have a esp32-2432S024C board, the capacitive touch version. Compiling the demo, The .elf is generated, however I get this linker error:
esp32_smartdisplay.c:240: undefined reference to `lvgl_touch_init'
Maybe I missed something?

**Comment by rzeldent** [comment]
Hi vorosj,

Strange, build passes: https://github.com/rzeldent/esp32-smartdisplay-demo/actions/runs/7597335667/job/20692040233
Please update the source code. Maybe something not up to date? Also do not forget to include/update submodules

**Reply by vorosj** [reply]
Thank you for the fast reply.

I write my steps, maybe helps.
I already have a working PIO on a Win11 machine, many succesful ESP32 projects etc.
I downloaded the esp32-smartdisplay-demo as a .zip, unzipped it.
Opened the unzipped project in PIO. 
Uncommented my board in the .ini file.
Copied the .json board description file into the PIO folder.
Then compiled the project.
Should I do any further steps?

The last few lines: 
Compiling .pio\build\esp32-2432S024C\FrameworkArduino\wiring_pulse.c.o
Compiling .pio\build\esp32-2432S024C\FrameworkArduino\wiring_shift.c.o
Archiving .pio\build\esp32-2432S024C\libFrameworkArduino.a
Linking .pio\build\esp32-2432S024C\firmware.elf
c:/users/mmormota/.platformio/packages/toolchain-xtensa-esp32@8.4.0+2021r2-patch5/bin/../lib/gcc/xtensa-esp32-elf/8.4.0/../../../../xtensa-esp32-elf/bin/ld.exe: .pio\build\esp32-2432S024C\lib126\libesp32_smartdisplay.a(esp32_smartdisplay.c.o):(.literal.smartdisplay_init+0x28): undefined reference to `lvgl_lcd_init'
c:/users/mmormota/.platformio/packages/toolchain-xtensa-esp32@8.4.0+2021r2-patch5/bin/../lib/gcc/xtensa-esp32-elf/8.4.0/../../../../xtensa-esp32-elf/bin/ld.exe: .pio\build\esp32-2432S024C\lib126\libesp32_smartdisplay.a(esp32_smartdisplay.c.o):(.literal.smartdisplay_init+0x2c): undefined reference to `lvgl_touch_init'
c:/users/mmormota/.platformio/packages/toolchain-xtensa-esp32@8.4.0+2021r2-patch5/bin/../lib/gcc/xtensa-esp32-elf/8.4.0/../../../../xtensa-esp32-elf/bin/ld.exe: .pio\build\esp32-2432S024C\lib126\libesp32_smartdisplay.a(esp32_smartdisplay.c.o): in function `smartdisplay_init':
C:\0work\yellow3\esp32-smartdisplay-demo-main/.pio/libdeps/esp32-2432S024C/esp32_smartdisplay/src/esp32_smartdisplay.c:233: undefined reference to `lvgl_lcd_init'
c:/users/mmormota/.platformio/packages/toolchain-xtensa-esp32@8.4.0+2021r2-patch5/bin/../lib/gcc/xtensa-esp32-elf/8.4.0/../../../../xtensa-esp32-elf/bin/ld.exe: C:\0work\yellow3\esp32-smartdisplay-demo-main/.pio/libdeps/esp32-2432S024C/esp32_smartdisplay/src/esp32_smartdisplay.c:240: undefined reference to `lvgl_touch_init'
collect2.exe: error: ld returned 1 exit status
*** [.pio\build\esp32-2432S024C\firmware.elf] Error 1
======================================================================= [FAILED] Took 58.41 seconds =======================================================================

Environment      Status    Duration
---------------  --------  ------------
esp32-2432S024C  FAILED    00:00:58.410
================================================================== 1 failed, 0 succeeded in 00:00:58.410 ================================================================== 

 *  The terminal process "C:\Users\mmormota\.platformio\penv\Scripts\platformio.exe 'run'" terminated with exit code: 1. 
 *  Terminal will be reused by tasks, press any key to close it.

**Comment by rzeldent** [comment]
Yes, you also need the boards directory as a git submodule.

**Reply by vorosj** [reply]
I tried the branches, and finally it works. The final mod:
In the demo, lib_deps includes the develop version: https://github.com/rzeldent/esp32-smartdisplay#develop and it failed.

Switched to main  and it works:
lib_deps =
    https://github.com/rzeldent/esp32-smartdisplay#main
    https://github.com/schreibfaul1/ESP32-audioI2S

(It's a bit odd syntax in PIO that # normally means comment in .ini, even it makes the word green - but not there.)

**Comment by rzeldent** [comment, marked as answer]
Correct, main demo was on the develop branch.
Is now corrected

Thanks [vorosj](https://github.com/vorosj)!
